src/core/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -279,29 +279,34 @@ QString QgsGdalProvider::dataSourceUri( bool expandAuthConfig ) const
 {
   if ( expandAuthConfig && QgsDataProvider::dataSourceUri( ).contains( QLatin1String( "authcfg" ) ) )
   {
-    QString uri( QgsDataProvider::dataSourceUri() );
-    // Check for authcfg
-    QRegularExpression authcfgRe( R"raw(authcfg='?([^'\s]+)'?)raw" );
-    QRegularExpressionMatch match;
-    if ( uri.contains( authcfgRe, &match ) )
-    {
-      uri = uri.replace( match.captured( 0 ), QString() );
-      QString configId( match.captured( 1 ) );
-      QStringList connectionItems;
-      connectionItems << uri;
-      if ( QgsApplication::authManager()->updateDataSourceUriItems( connectionItems, configId, QStringLiteral( "gdal" ) ) )
-      {
-        uri = connectionItems.first( );
-      }
-    }
-    return uri;
+    return QgsGdalProvider::expandAuthConfig( QgsDataProvider::dataSourceUri() );
   }
   else
   {
     return QgsDataProvider::dataSourceUri();
   }
 }
 
+QString QgsGdalProvider::expandAuthConfig( const QString &dsName )
+{
+  QString uri( dsName );
+  // Check for authcfg
+  QRegularExpression authcfgRe( " authcfg='([^']+)'" );
+  QRegularExpressionMatch match;
+  if ( uri.contains( authcfgRe, &match ) )
+  {
+    uri = uri.replace( match.captured( 0 ), QString() );
```

Comment:
Maybe better: https://doc.qt.io/qt-5/qstring.html#remove-4

Reply:
@elpaso , I'd rather stick to using the QRegularExpressionMatch we already have (and must have to fetch the config ID) instead of what I assume is an extra regular expression matching when using QString::remove( QRegularExpression &re ). It avoids the microscopic cost associated to running the regexp engine twice :)

Reply:
Then why not https://doc.qt.io/qt-5/qstring.html#remove-3 ?

Reply:
@elpaso , because the code was cherry picked from https://github.com/qgis/QGIS/blob/master/src/core/providers/ogr/qgsogrproviderutils.cpp#L2111 and came that way. I've got no strong feelings about it, but can't much of a difference here.